presto-main/src/main/java/com/facebook/presto/execution/SetPathTask.java

```diff
@@ -53,6 +54,10 @@ public String getName()
     {
         Session session = stateMachine.getSession();
 
+        if (!session.getClientCapabilities().contains("path")) {
```

Comment:
should not this be something checked at the lower level, closer to the communication? When state machine have path set then we could check if client is capable to handle the path before sending him a response. What do you think?

Here it is simple, but in case of roles (https://github.com/prestodb/presto/pull/10904) we would need to check in many places as there are many role related statements. Would we need to update all of these statements execution code? I would prefer to change a single place instead.

Also here you are mixing domains: SQL execution with client communication.

Reply:
We don't really have a way to do so - we want to fail the query itself, so if we put the check later (eg in StatementResource), then the query will have completed and the error won't behave as we wanted.